src/engine/MILPEncoder.cpp

```diff
@@ -118,11 +118,70 @@ void MILPEncoder::encodeReLUConstraint( GurobiWrapper &gurobi, ReluConstraint
     List<GurobiWrapper::Term> terms;
     terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", targetVariable ) ) );
     terms.append( GurobiWrapper::Term( -1, Stringf( "x%u", sourceVariable ) ) );
-    terms.append( GurobiWrapper::Term( -sourceLb, Stringf( "a%u", index ) ) );
+    terms.append( GurobiWrapper::Term( -sourceLb, Stringf( "a%u", _binVarIndex ) ) );
     gurobi.addLeqConstraint( terms, -sourceLb );
 
     terms.clear();
     terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", targetVariable ) ) );
-    terms.append( GurobiWrapper::Term( -sourceUb, Stringf( "a%u", index ) ) );
+    terms.append( GurobiWrapper::Term( -sourceUb, Stringf( "a%u", _binVarIndex++ ) ) );
     gurobi.addLeqConstraint( terms, 0 );
 }
+
+void MILPEncoder::encodeMaxConstraint( GurobiWrapper &gurobi, MaxConstraint *max )
+{
+    if ( !max->isActive() )
+        return;
+
+    // y = max(x_1, x_2, ... , x_m)
+    unsigned y = max->getF();
+
+    // xs = [x_1, x_2, ... , x_m]
+    List<unsigned> xs = max->getElements();
+
+    // upper bounds of each x_i
+    using qtype = std::pair<double, unsigned>;
+    auto cmp = []( qtype l, qtype r) { return l.first <= r.first; };
+    std::priority_queue<qtype, std::vector<qtype>, decltype( cmp )> ubq( cmp );
+
+    // terms for Gurobi
+    List<GurobiWrapper::Term> terms;
+
+    for ( const auto &x : xs ) 
+    {
+        // add binary variable
+        gurobi.addVariable( Stringf( "a%u_%u", _binVarIndex, x ),
```

Comment:
Please explain the name convention for future reference.

Reply:
I did.